plugins/modem-manager/fu-plugin-modem-manager.c

```diff
@@ -414,10 +413,16 @@ fu_plugin_update_attach (FuPlugin *plugin, FuDevice *device, GError **error)
 	if (locker == NULL)
 		return FALSE;
 
-	/* must be done in idle so that the engine explicitly
-	 * waits for the device to be redetected */
-	g_idle_add (fu_plugin_mm_uninhibit_device_idle, plugin);
+	/* schedule device attach asynchronously, which is extremely important
+         * so that engine can setup the device "waiting" logic before the actual
+         * attach procedure happens (which will reset the module if it worked
+         * properly) */
+	if (!fu_device_attach (FU_DEVICE (device), error))
```

Comment:
I don't think you need the `FU_DEVICE` here.

Reply:
True, will remove that.